google/cloud/internal/background_threads_impl_test.cc

```diff
@@ -72,12 +74,44 @@ TEST(CustomerSuppliedBackgroundThreads, SharesCompletionQueue) {
 /// @test Verify that automatically created completion queues are usable.
 TEST(AutomaticallyCreatedBackgroundThreads, IsActive) {
   AutomaticallyCreatedBackgroundThreads actual;
+  EXPECT_EQ(1, actual.pool_size());
 
   promise<std::thread::id> bg;
   actual.cq().RunAsync([&bg] { bg.set_value(std::this_thread::get_id()); });
   EXPECT_NE(std::this_thread::get_id(), bg.get_future().get());
 }
 
+/// @test Verify that automatically created completion queues are usable.
+TEST(AutomaticallyCreatedBackgroundThreads, NoEmptyPools) {
+  AutomaticallyCreatedBackgroundThreads actual(0);
+  EXPECT_EQ(1, actual.pool_size());
+
+  promise<std::thread::id> bg;
+  actual.cq().RunAsync([&bg] { bg.set_value(std::this_thread::get_id()); });
+  EXPECT_NE(std::this_thread::get_id(), bg.get_future().get());
+}
+
+/// @test Verify that automatically created completion queues work.
+TEST(AutomaticallyCreatedBackgroundThreads, ManyThreads) {
+  auto constexpr kThreadCount = 4;
+  AutomaticallyCreatedBackgroundThreads actual(kThreadCount);
+  EXPECT_EQ(kThreadCount, actual.pool_size());
+
+  std::vector<promise<std::thread::id>> promises(100 * kThreadCount);
+  for (auto& p : promises) {
+    actual.cq().RunAsync([&p] {
+      p.set_value(std::this_thread::get_id());
+      // Keep the thread busy for a bit so the other functors schedule in a
+      // different thread.
+      std::this_thread::sleep_for(std::chrono::milliseconds(1));
+    });
+  }
+  std::set<std::thread::id> ids;
+  for (auto& p : promises) ids.insert(p.get_future().get());
+  EXPECT_LE(2, ids.size());
```

Comment:
While this is a good objective for a four-thread pool, it may be flaky, and also seems beside the point of "nothing run on the main thread."

Reply:
Fair enough. Fixed.  FWIW, I did ran it thousands of times to avoid flakiness.